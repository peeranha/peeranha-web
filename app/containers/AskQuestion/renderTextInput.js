import React from 'react';
import PropTypes from 'prop-types';

import messages from './messages';

function renderTextInput({
  input,
  translations,
  disabled,
  meta: { touched, error, warning },
}) {
  return (
    <div>
      <h6>{translations[messages.titleLabel.id]}</h6>
      <input
        {...input}
        disabled={disabled}
        placeholder={translations[messages.titleLabel.id]}
        type="text"
        className="form-control"
      />
      <h6 className="text-danger">
        {touched &&
          ((error && <span>{translations[error]}</span>) ||
            (warning && <span>{translations[warning]}</span>))}
      </h6>
    </div>
  );
}

renderTextInput.propTypes = {
  input: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,
  disabled: PropTypes.bool,
  sendProps: PropTypes.object,
  translations: PropTypes.object,
};

export default renderTextInput;
