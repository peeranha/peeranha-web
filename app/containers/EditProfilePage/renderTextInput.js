import React from 'react';
import PropTypes from 'prop-types';

function renderTextInput({
  input,
  label,
  sendProps,
  meta: { touched, error, warning },
}) {
  return (
    <div>
      <h6>{label}</h6>
      <input
        {...input}
        placeholder={label}
        type="text"
        className="form-control"
      />
      <h6 className="text-danger">
        {touched &&
          ((error && <span>{sendProps.translations[error]}</span>) ||
            (warning && <span>{sendProps.translations[warning]}</span>))}
      </h6>
    </div>
  );
}

renderTextInput.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  sendProps: PropTypes.object.isRequired,
};

export default renderTextInput;
