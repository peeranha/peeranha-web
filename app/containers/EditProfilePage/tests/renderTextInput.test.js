import renderer from 'react-test-renderer';
import 'jest-styled-components';

import renderTextInput from '../renderTextInput';

describe('renderTextInput test', () => {
  it('test by snapshots', () => {
    const renderedComponent = renderer
      .create(
        renderTextInput({
          input: {},
          label: 'string',
          sendProps: {},
          meta: {},
        }),
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
