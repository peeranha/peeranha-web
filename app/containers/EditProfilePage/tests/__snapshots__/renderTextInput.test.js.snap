// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderTextInput test test by snapshots 1`] = `
<div>
  <h6>
    string
  </h6>
  <input
    className="form-control"
    placeholder="string"
    type="text"
  />
  <h6
    className="text-danger"
  />
</div>
`;
